plugins/docker/attacher.go

```diff
@@ -137,12 +216,36 @@ func (m *AttachManager) attach(id string) {
 	fields["ContainerID"] = id
 	fields["ContainerName"] = name
 
+	// NOTE! Anything that is a duplicate key will be overridden with the value
+	// that is in the label, not the environment
+	for _, key := range m.fieldsFromLabels {
+		if value, ok := container.Config.Labels[key]; ok {
+			fields[key] = value
+		}
+	}
+
+	return fields, nil
+}
+
+// Attach to the output of a single running container
+func (m *AttachManager) attach(id string, client DockerClient) error {
+	m.ir.LogMessage(fmt.Sprintf("Attaching container: %s", id))
+
+	fields, err := m.extractFields(id, client)
+	if err != nil {
+		m.ir.LogError(err)
+		return err
```

Comment:
That's reasonable. There was no error logging before, so hey, it's better than that. I'll take a pass through it and consider where it might be duplicated.